### PR TITLE
toy touch point environment for option learning

### PR DESCRIPTION
**Pull request by tomsilver**
example with random actions:
https://user-images.githubusercontent.com/2816502/161387887-a4d97626-5993-4392-b08b-81c3b6cf7cce.mp4

example with learned nsrts:
https://user-images.githubusercontent.com/2816502/161387868-e5ad0330-f371-467a-921d-900851d056b6.mp4

things that I tested:
* random actions
* random options
* oracle
* nsrt learning

### CODE REVIEW

File: src/envs/touch_point.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    def types(self) -> Set[Type]:
+        return {self._robot_type, self._target_type}
+
+    @property
+    def options(self) -> Set[ParameterizedOption]:
+        return {self._MoveTo}
+
+    @property
+    def action_space(self) -> Box:
+        # An angle in radians. Wraps around to avoid boundary issues.
```
Comment (ronuchit) [inline src/envs/touch_point.py:97]: what part of the code is handling the boundary issues of the angle?

Comment (ronuchit) [reply src/envs/touch_point.py:97]: the move policy seems to be just -pi to pi?

File: src/envs/touch_point.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        self._robot_type = Type("robot", ["x", "y"])
+        self._target_type = Type("target", ["x", "y"])
+        # Predicates
+        self._Touched = Predicate("Touched",
+                                  [self._robot_type, self._target_type],
+                                  self._Touched_holds)
+        # Options
+        self._MoveTo = ParameterizedOption(
+            "MoveTo",
+            types=[self._robot_type, self._target_type],
```
Comment (ronuchit) [inline src/envs/touch_point.py:46]: so plans are just length 1: MoveTo(robot, target), always?

File: src/settings.py [hunk 3]
```diff
@@ -245,6 +245,9 @@ def get_arg_specific_settings(args: Dict[str, Any]) -> Dict[str, Any]:
                     # lower level, so tasks take more actions to complete.
                     "behavior": 1000,
                     "pybullet_blocks": 1000,
+                    # For the very simple TouchPoint environment, restrict
+                    # the horizon to be shorter.
+                    "touch_point": 15,
```
Comment (ronuchit) [inline src/settings.py:250]: is this the horizon at the action level or the option level?

File: tests/envs/test_touch_point.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            "x": 0.5,
+            "y": 0.9,
+        }
+    })
+    assert len(task.goal) == 1
+    goal_atom = list(task.goal)[0]
+    assert goal_atom.predicate.name == "Touched"
+    assert goal_atom.objects == [robot, target]
+    assert not goal_atom.holds(state)
+    action = Action(np.array([np.pi / 2], dtype=np.float32))  # move up
```
Comment (ronuchit) [inline tests/envs/test_touch_point.py:47]: if actions are -2*pi to 2*pi, are there actually two angles you could use here to move up? should we eliminate that redundancy?

**Review by ronuchit — APPROVED** [review]
lgtm, questions

File: src/envs/touch_point.py (hunk 1 above)
Comment (tomsilver) [reply src/envs/touch_point.py:97]: yeah nvm I don't think this is necessary after all

File: src/envs/touch_point.py (hunk 2 above)
Comment (tomsilver) [reply src/envs/touch_point.py:46]: yeah

File: src/settings.py (hunk 3 above)
Comment (tomsilver) [reply src/settings.py:250]: action level

**Pull request merged**